modules/oe_list_pages_open_vocabularies/src/SearchApiConfigurator.php

```diff
@@ -0,0 +1,186 @@
+
+    $index_field->setPropertyPath($property_path);
+    $index_field->setDatasourceId('entity:' . $entity_type);
+    $index_field->setLabel($association->label());
+    $index->save();
+
+    // Generate the facet id.
+    $id = 'open_vocabularies_' . str_replace('.', '_', $association->id() . '_' . $field_id);
+    // Create the facet for the new field.
+    /** @var \Drupal\facets\FacetInterface $facet */
+    $facet = $this->getFacet($id);
+    $facet->setUrlAlias(str_replace('.', '_', $association->id()));
+    $facet->set('name', $association->label());
+    $facet->setOnlyVisibleWhenFacetSourceIsVisible(TRUE);
+    $facet->setWeight(0);
+    $facet->addProcessor([
+      'processor_id' => 'display_value_widget_order',
+      'weights' => [],
+      'settings' => [
+        'sort' => 'ASC',
+      ],
+    ]);
+    $facet->addProcessor([
+      'processor_id' => 'url_processor_handler',
+      'weights' => [],
+      'settings' => [],
+    ]);
+    $facet->addProcessor([
+      'processor_id' => 'translate_entity',
+      'weights' => [],
+      'settings' => [],
+    ]);
+    $facet->setEmptyBehavior(['behavior' => 'none']);
+    $facet->setFacetSourceId($list_source->getSearchId());
+    $facet->setWidget('oe_list_pages_multiselect', []);
+    $facet->setFieldIdentifier($property_path);
+    $facet->save();
+  }
+
+  /**
+   * Remove search index field and facet created for an association.
+   *
+   * @param \Drupal\open_vocabularies\OpenVocabularyAssociationInterface $association
+   *   The association configuration.
+   * @param string $field_id
+   *   The field id.
+   */
+  public function removeConfig(OpenVocabularyAssociationInterface $association, string $field_id): void {
+    // Based on the list source, determine the correct index to create the
+    // field in.
+    $field_config = $this->entityTypeManager->getStorage('field_config')->load($field_id);
+    $entity_type = $field_config->getTargetEntityTypeId();
+    $bundle = $field_config->getTargetBundle();
+    /** @var \Drupal\oe_list_pages\ListSourceInterface $list_source */
+    $list_source = $this->listSourceFactory->get($entity_type, $bundle);
+    /** @var \Drupal\search_api\IndexInterface $index */
+    $index = $list_source->getIndex();
+
+    // Remove field.
+    $id = 'open_vocabularies_' . str_replace('.', '_', $association->id() . '_' . $field_id);
+    $index->removeField($id);
+    $index->save();
+
+    // Remove facet.
+    $facet = $this->getFacet($id);
+    $facet->delete();
+  }
+
+  /**
+   * Gets an existing facet by id or creat a new one if doesn't exist.
+   *
+   * @param string $facet_id
+   *   The facet id.
+   *
+   * @return \Drupal\facets\FacetInterface
+   *   The facet.
+   */
+  protected function getFacet(string $facet_id): FacetInterface {
+    $facet = $this->entityTypeManager->getStorage('facets_facet')->load($facet_id);
+    return $facet instanceof FacetInterface ? $facet : Facet::create(['id' => $facet_id]);
```

Comment:
Aa, here we should use the storage and not call Facet statically...i missed it. And by missed it, I mean, I introduced it :D